### PR TITLE
cleanup sql package, move submitters to executor pkg

### PR DESCRIPTION
**Pull request by Yancey0623**
## Summary
This PR tried to cleanup the `sql` package, move `submitter` code to `executor` package.

### `Executor` and `Submitter` concept confusion.

The `codegen` package generates Python code,`pai/alisa/defaultSubmitter` executes the generated code by PAI command-line tool, alias RPC service, or Python interrupt. To look at the implementation, there is an `Executor` interface that aims to execute SQL Statement, and and `submitter` interface embedded `Executor` interface and named to `xxxSubmitter`. I think we only need the `Executor` concept.

https://github.com/sql-machine-learning/sqlflow/blob/f5dc0209fe1bd71c443e82ebb4f6981b06e33542/pkg/ir/ir.go#L18-L25

https://github.com/sql-machine-learning/sqlflow/blob/f5dc0209fe1bd71c443e82ebb4f6981b06e33542/pkg/sql/submitter.go#L66-L70

### `sql` package does too many things.

the `sql` package includes IR generator, code generator and executor(submitter), we should decompose them into individual pacakges.

## Solution

1. Refactor submitter code to executor which implement the `Executor` interface.
1. A new package named `executor` separated from `sql` pacakge.

### CODE REVIEW

**Comment by shendiaomo** [comment]
Great! There're 2 suggestions from me:
1. We revert the relationship of the visitor architecture. Because we add many more statements than platforms.
1. Decouple the executor and the codegen

File: pkg/executor/pre_exec.go [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+		_, fld := verifier.Decomp(ct.Name())
+		typeName := ct.DatabaseTypeName()
+		flds = append(flds, fld)
+		ft = append(ft, typeName)
+	}
+
+	return flds, ft, nil
+}
+
+func doAttrInitAndTypeChecking(ir *ir.TrainStmt) error {
```
Comment (wangkuiyi) [inline pkg/executor/pre_exec.go]: InitializeAndCheckAttributes ?

File: pkg/executor/pre_exec.go [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	createStmt := b.String()
+	if _, e := db.Exec(createStmt); e != nil {
+		return fmt.Errorf("failed executing %s: %q", createStmt, e)
+	}
+	return nil
+}
+
+// getColumnTypes is quiet like verify but accept a SQL string as input, and returns
+// an ordered list of the field types.
+func getColumnTypes(slct string, db *database.DB) ([]string, []string, error) {
```
Comment (wangkuiyi) [inline pkg/executor/pre_exec.go]: getSQLFieldType

File: pkg/executor/pre_exec.go [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+	pb "sqlflow.org/sqlflow/pkg/proto"
+	"sqlflow.org/sqlflow/pkg/sql/codegen/pai"
+	"sqlflow.org/sqlflow/pkg/sql/codegen/tensorflow"
+	"sqlflow.org/sqlflow/pkg/sql/codegen/xgboost"
+	"sqlflow.org/sqlflow/pkg/step/feature"
+	"sqlflow.org/sqlflow/pkg/verifier"
+)
+
+// Create prediction table using the `PredictStmt`.
+func createPredictionTableFromIR(predStmt *ir.PredictStmt, db *database.DB, session *pb.Session) error {
```
Comment (wangkuiyi) [inline pkg/executor/pre_exec.go]: createPredictionResultTable?

File: pkg/executor/pre_exec.go [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+	// the result column type by the prediction result.
+	//
+	// label column not found in predict table, create a column specified by PREDICT clause:
+	if resultColumnType == "" {
+		// NOTE(typhoonzero): Clustering model may not have label in select statement, default use INT type
+		resultColumnType = "INT"
+	}
+	stype, e := fieldType(db.DriverName, resultColumnType)
+	if e != nil {
+		return e
```
Comment (wangkuiyi) [inline pkg/executor/pre_exec.go]: Need explanation here in addition to returning e

**Review by wangkuiyi — COMMENTED** [review]
LGTM with some minor naming suggestions.

**Review by shendiaomo — APPROVED** [review]

**Pull request merged**